CONTRIBUTING.md

```diff
@@ -1,88 +1,203 @@
-* DO provide strongly typed APIs.
-* DO use verbose identifier names.
-
-## Files and folders
-
-* DO associate no more than one class per file.
-* DO use folders to group classes based on features.
+You can contribute to this project by contributing to:
+
+* [Issues](https://github.com/microsoft/TemplateStudio/issues)
+* [Discussions](https://github.com/microsoft/TemplateStudio/discussions)
+* [Templates](#Templates)
+
+If you intend to contribute code changes, learn how to [set up your development environment](#Set-up-your-development-environment).
+
+When contributing template changes, [validate](#Validating-changes) your changes by generating projects with updated templates, then file a PR to trigger CI validation.
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
+contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+## Set up your development environment
+
+* Install [Visual Studio 2022 (.NET Desktop and Visual Studio Extension Workloads)](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community&channel=Release&version=VS2022&source=VSLandingPage&passive=false)
+* Install [Visual Studio Code](https://code.visualstudio.com/Download)
+* Clone the [repo](https://github.com/microsoft/TemplateStudio.git)
+
+## Extensions
+
+[TemplateStudio.sln](https://github.com/microsoft/TemplateStudio/blob/main/code/TemplateStudio.sln) is a multi-project solution that contains all of the extension projects and associated tests and assets. Each extension lives under a corresponding `TemplateStudioFor*` project in this solution.
+
+There are also `*.slnf` files that load the subset of projects needed for a specific extension if you want a more focused and potentially more performant workspace:
+
+* [TemplateStudio.WinUICs.slnf](https://github.com/microsoft/TemplateStudio/blob/main/code/TemplateStudio.WinUICs.slnf)
+* [TemplateStudio.WinUICpp.slnf](https://github.com/microsoft/TemplateStudio/blob/main/code/TemplateStudio.WinUICpp.slnf)
+* [TemplateStudio.WPF.slnf](https://github.com/microsoft/TemplateStudio/blob/main/code/TemplateStudio.WPF.slnf)
+* [TemplateStudio.UWP.slnf](https://github.com/microsoft/TemplateStudio/blob/main/code/TemplateStudio.UWP.slnf)
+
+Regardless of the solution you load, you need to make sure the extension project you want to work on is set as the startup project, and you need to ensure you select an appropriate Debug or Release configuration specific to that extension (e.g. `DebugWinUICs`/`ReleaseWinUICs`).
+
+When you F5 to debug or start without debugging, Visual Studio will install the extensions in the Visual Studio Experimental Instance. Once loaded, you can test the extensions by creating a new project and selecting the appropriate `Template Studio for *` template.
+
```

Comment:
Also mention using VSEmulator to make testing the wizard faster and easier.

Reply:
Added.